core/condition.go

```diff
@@ -397,18 +397,17 @@ func blockNum(result *Result, first, second string) bool {
 		return false
 
 	}
-
-	result.AddError(fmt.Sprintf("rpc: %d, reference: %d", firstI, secondI))
-	return !(absInt(secondI-firstI) > deltaThreshold)
+	result.AddError(fmt.Sprintf("rpc: %d, reference: %d, diff: %d", firstI, secondI, secondI-firstI))
+	return !(secondI-firstI > deltaThreshold)
```

Comment:
Please revert the change on this line. We must use an absolute comparison, otherwise it doesn't make sense to have a delta comparison.

See commit 3ca8636e6eb9d1cdec772f5cb92c028a582730ab.

Reply:
Why do we need an absolute comparison? If we keep it as it is, it will continue to alert when the public RPC is behind our node.

Reply:
Here's the situation we are dealing with: we have two nodes of a distributed system. Both are advancing in state, each one in _their own rhythm_.

We want to make sure that at any given point in time, both nodes "agree" to each other. But it is impossible to guarantee that they will always be in perfect sync, so we allow a small difference (i.e. `deltaThreshold` in the code) between them. Whenever we talk about distributed systems, we expect this small difference in state. And it is fine, as long as it is a _small_ difference. It does not matter which node is ahead, that's why we need an absolute comparison.

By removing the `absInt()` function, whenever the `reference` is ahead of our `rpc` by `1` block (regardless of the `deltaThreshold` value), the check will fail and generate an alert.

Distributed systems are very difficult to understand and tame, and even the smaller code changes can have a huge impact in the overall machinery. Ping me if you want to go deeper into this, I can gladly talk about it for hours :)

Reply:
The theoretical part is clear to me, but in practice, I'm missing why we want to receive notifications when the public RPC (reference) is behind our node. Note that I tested this change and all scenario tests passed. In addition, it will reduce false alarms.